22.py

```diff
@@ -0,0 +1,15 @@
+class Solution:
+    def generateParenthesis(self, n: int) -> List[str]:
+        return self.genParanths(n, 0, 0)
+    
+    def genParanths(self, n: int, used_opens: int, used_closes: int, prefix="") -> List[str]:
```

Comment:
It's elegant to have the default argument for `prefix=""` but I think it would be safer to not make it default and then call `""` in the first call

Reply:
It's more explicit this way and reduces the chance of unintended side effects